dockers/Screenshotter/screenshotter.js

```diff
@@ -463,3 +506,21 @@ function promisify(f) {
     f.apply(null, args);
     return deferred.promise;
 }
+
+// Execute a given command, and return a promise to its output.
+// Don't denodeify here, since fail branch needs access to stderr.
+function execFile(cmd, args, opts) {
+    const deferred = new selenium.promise.Deferred();
+    childProcess.execFile(cmd, args, opts, function(err, stdout, stderr) {
+        if (err) {
+            console.error("Error executing " + cmd + " " + args.join(" "));
+            console.error(stdout + stderr);
+            err.stdout = stdout;
+            err.stderr = stderr;
+            deferred.reject(err);
+        } else {
+            deferred.fulfill(stdout);
+        }
+    });
+    return deferred.promise;
+}
```

Comment:
It's definitely preferable to to have these as separate commits.  Even though it's all going to the same place, smaller commits will take less time to review and if one review needs revision it doesn't hold up the other.